### PR TITLE
CreateIfNotExists operator in Patch operations

### PR DESCRIPTION
**Pull request by balalakshmin**
# Description

CreateIfNotExists operator will first check if all the segments in the path provided (in the request) exist in the document.
In case they do not, the operator will create them and set the provided value at the leaf segment of the provided path. In case the path already exists, the operator will not perform any modifications to the existing document at all.

To add or update an optional field, use createIfNotExists followed by add/set
## Sample
```
CosmosPatchOperations cosmosPatch = CosmosPatchOperations.create();
// will add the path /g if it doesn't exist and initialize it with []. Will be a no-op if /g already exists
cosmosPatch.createIfNotExists("/g", []); 
// will set /g with the desired content
cosmosPatch.set("/g", [0,1,2]); 
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/patch/PatchOperationType.java [hunk 1]
```diff
@@ -10,7 +10,8 @@ public enum PatchOperationType {
     REPLACE(PatchConstants.OperationTypeNames_Replace),
     MOVE(PatchConstants.OperationTypeNames_Move),
     SET(PatchConstants.OperationTypeNames_Set),
-    INCREMENT(PatchConstants.OperationTypeNames_Increment);
+    INCREMENT(PatchConstants.OperationTypeNames_Increment),
+    CREATE_IF_NOT_EXISTS(PatchConstants.OperationTypeNames_CreateIfNotExists);
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/patch/PatchOperationType.java:14]: NIT - align indentation

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
The new patch operation also needs to be added here - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPatchHelper.scala

And test coverage from Spark for the new patch operation needs to be added as well.

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 #### Features Added
 * Integrate `ThroughputControl` with ChangeFeedProcessor - See [PR 38052](https://github.com/Azure/azure-sdk-for-java/pull/38052)
+* Added `CreateIfNotExists` operator in PatchOperations - See [PR 38173](https://github.com/Azure/azure-sdk-for-java/pull/38173)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:7]:
```suggestion
* Added `createIfNotExists` operator in PatchOperations - See [PR 38173](https://github.com/Azure/azure-sdk-for-java/pull/38173)
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java [hunk 3]
```diff
@@ -291,6 +291,48 @@ public CosmosPatchOperations increment(String path, double value) {
...
+    /**
+     * The operator will first check if all the segments in the path provided (in the request) exist in the document.
+     * In case they do not, the operator will create them and set the provided value at the leaf segment of the provided path.
+     * In case the path already exists, the operator will not perform any modifications to the existing document at all.
+     *
+     * For the above JSON, we can have something like this:
+     * <code>
+     *     CosmosPatchOperations cosmosPatch = CosmosPatchOperations.create();
+     *     cosmosPatch.createIfNotExists("/a/x/y", 15); // will add the path /a/x/y and set the value to 15
+     *     cosmosPatch.createIfNotExists("/a/b/c", "new value"); // will not perform any modification as the path already exists
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java:303]: The documentation says if all the segments in the path provided exist. however, in this case, /b/c does not exist. Is my understanding incorrect?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
@balalakshmin thanks for the changes, however, we should add some more tests covering the multiple path scenarios that we have mentioned in the java docs.

**Pull request closed without merging**